libs/parallelism/execution/include/hpx/execution/algorithms/let_error.hpp

```diff
@@ -0,0 +1,284 @@
+                    {
+                        hpx::execution::experimental::set_done(std::move(r));
+                    };
+
+                    template <typename... Ts>
+                    void set_value(Ts&&... ts) noexcept
+                    {
+                        hpx::execution::experimental::set_value(
+                            std::move(r), std::forward<Ts>(ts)...);
+                    }
+                };
+
+                // Type of the operation state returned when connecting the
+                // predecessor sender to the let_error_predecessor_receiver
+                using predecessor_operation_state_type = std::decay_t<
+                    connect_result_t<PS&&, let_error_predecessor_receiver>>;
+
+                // Type of the potential operation states returned when
+                // connecting a sender in successor_sender_types to the receiver
+                // connected to the let_error_sender
+                template <typename Sender>
+                struct successor_operation_state_types_helper
+                {
+                    using type = connect_result_t<Sender, R>;
+                };
+                template <template <typename...> class Variant>
+                using successor_operation_state_types =
+                    hpx::util::detail::transform_t<
+                        successor_sender_types<Variant>,
+                        successor_operation_state_types_helper>;
+
+                // Operation state from connecting predecessor sender to
+                // let_error_predecessor_receiver
+                predecessor_operation_state_type predecessor_os;
+
+                // Potential errors returned from the predecessor sender
+                hpx::util::detail::prepend_t<
+                    predecessor_error_types<std::variant>, std::monostate>
+                    predecessor_e;
+
+                // Potential operation states returned when connecting a sender
+                // in successor_sender_types to the receiver connected to the
+                // let_error_sender
+                hpx::util::detail::prepend_t<
+                    successor_operation_state_types<std::variant>,
+                    std::monostate>
+                    successor_os;
+
+                template <typename PS_, typename R_, typename F_>
+                operation_state(PS_&& ps, R_&& r, F_&& f)
+                  : predecessor_os{hpx::execution::experimental::connect(
+                        std::forward<PS_>(ps),
+                        let_error_predecessor_receiver(
+                            std::forward<R_>(r), std::forward<F_>(f), *this))}
+                {
+                }
+
+                operation_state(operation_state&&) = delete;
+                operation_state& operator=(operation_state&&) = delete;
+                operation_state(operation_state const&) = delete;
+                operation_state& operator=(operation_state const&) = delete;
```

Comment:
Sometimes the `operation_state` is made non-copyable, sometimes not. Is there a pattern?

Reply:
In general they should be non-copyable, and non-movable (or rather, they can be copyable and movable, but to assume that they are would be an error). I see I've missed at least one spot. Did you notice many places where copy/move constructors/assignment operators are not deleted?

Reply:
I've updated these now to always be non-movable and non-copyable.